src/dsp_tools/utils/excel_to_json/utils.py

```diff
@@ -0,0 +1,296 @@
+    """
+    This function takes a pd.Series which has "comment_[language tag]" in the index.
+    If the value of the index is not pd.NA, the language tag and the value are added to a dictionary.
+    If it is empty, it is omitted from the dictionary.
+
+    Args:
+        df_row: a pd.Series (usually a row of a pd.DataFrame) from which the content of the columns containing the
+                comment is extracted
+
+    Returns:
+        A dictionary with the language tag and the content of the cell
+    """
+    comments = {lang: df_row[f"comment_{lang}"] for lang in languages if df_row[f"comment_{lang}"] is not pd.NA}
+    if comments == {}:
+        return None
+    else:
+        return comments
+
+
+def find_one_full_cell_in_cols(df: pd.DataFrame, required_columns: list[str]) -> pd.Series | None:
+    """
+    This function takes a pd.DataFrame and a list of column names where at least one cell must have a value per row.
+    It creates a pd.Series with boolean values that are True if the cell is empty for each column.
+    These series are then combined.
+    In the resulting np.array the values are only True if all the values from the pd.Series were True,
+    meaning that in this row, all the specified columns have no values.
+    If any of the values in the np.array are True,
+    it converts it into a pd.Series (the data type is relevant at a later point) and returns it.
+    If all the values are False, then it returns None.
+
+    Args:
+        df: The pd.DataFrame which should be checked
+        required_columns: A list of column names where at least one cell per row must have a value
+
+    Returns:
+        None if there is no problem or a pd.Series if there is a problem in a row
+    """
+    # In order to combine more than two arrays, we need to reduce the arrays, which takes a tuple
+    result_arrays = tuple(df[col].isnull() for col in required_columns)
+    # If all are True logical_and returns True otherwise False
+    combined_array = np.logical_and.reduce(result_arrays)
+    if any(combined_array):
+        return pd.Series(combined_array)
+    else:
+        return None
+
+
+def col_must_or_not_empty_based_on_other_col(
+    df: pd.DataFrame,
+    substring_list: list[str],
+    substring_colname: str,
+    check_empty_colname: str,
+    must_have_value: bool,
+) -> pd.Series | None:
+    """
+    It is presumed that the column "substring_colname" has no empty cells.
+    Based on the string content of the individual rows, which is specified in the "substring_list",
+    the cell is the column "check_empty_colname" is checked whether it is empty or not.
+    The "substring_list" contains the different possibilities regarding the content of the cell.
+    They are joined in a RegEx "|" which denotes "or".
+    If it does not match any of the sub-strings, then the RegEx returns
+    False which means that the value in the column "check_empty_colname" is not relevant.
```

Comment:
Docstrings should explain to users what can be expected from this function, not how the function works internally. And we always have to assume that users have only access to the docstring, but not to the code. (DaSCH is open source, but imagine a closed source project: There, the user has only the docstring, but not the code.)

For the user of this function, it can be frustrating to read about return values of regexes, because s/he is not in direct contact with these regexes. S/he is only interested in:

1. What can I feed into the function? (input)
2. What will the function give me? (output)

Everything in between is implementation detail, that should be hidden from the user!